src/test/features/response/routes/when-did-you-pay.ts

```diff
@@ -0,0 +1,115 @@
+import { expect } from 'chai'
+import * as request from 'supertest'
+import * as config from 'config'
+
+import { attachDefaultHooks } from '../../../routes/hooks'
+import '../../../routes/expectations'
+import { checkAuthorizationGuards } from './checks/authorization-check'
+import { checkAlreadySubmittedGuard } from './checks/already-submitted-check'
+
+import { Paths as ResponsePaths } from 'response/paths'
+
+import { app } from '../../../../main/app'
+
+import * as idamServiceMock from '../../../http-mocks/idam'
+import * as draftStoreServiceMock from '../../../http-mocks/draft-store'
+import * as claimStoreServiceMock from '../../../http-mocks/claim-store'
+import { checkNotDefendantInCaseGuard } from './checks/not-defendant-in-case-check'
+
+const cookieName: string = config.get<string>('session.cookieName')
+
+const pagePath = ResponsePaths.whenDidYouPay.evaluateUri({ externalId: claimStoreServiceMock.sampleClaimObj.externalId })
+describe('Defendant response: when did you pay', () => {
+  attachDefaultHooks()
+
+  describe('on GET', () => {
+    const method = 'get'
+    checkAuthorizationGuards(app, method, pagePath)
+    checkNotDefendantInCaseGuard(app, method, pagePath)
+
+    context('when user authorised', () => {
+      beforeEach(() => {
+        idamServiceMock.resolveRetrieveUserFor(claimStoreServiceMock.sampleClaimObj.defendantId, 'citizen')
+      })
+
+      checkAlreadySubmittedGuard(app, method, pagePath)
+
+      context('when response not submitted', () => {
+        it('should return 500 and render error page when cannot retrieve claim', async () => {
```

Comment:
Please add another test case simulating failure in draft retrieval. Do the same for POST handler as well.